nix-store.el

```diff
@@ -22,5 +28,152 @@ PATH the path within /nix/store to realise"
    :noquery t
    :name (format "*nix-store*<%s>" path)))
 
+(defvar-local nix-buffer-store-path nil "Buffer-local object holding an `nix-store-path` object")
+
+(defclass nix-store-path ()
+  ((path :initarg :path :accessor nix-store-path-path)
+    (status :initarg :status :initform 'realised)
+    (hash :initarg :hash)
+    (size :initarg :size)
+    (derivers :initarg :derivers)
+    (references :initarg :references)
+    (referrers :initarg :referrers)
+    (requisites :initarg :requisites))
+  "Nix-Store-Path Class that holds all information of the path that
+is displayed")
+
+(cl-defmethod nix-store-fill-data ((object nix-store-path))
+  (oset object :size (nix-store--query 'size (nix-store-path-path object)))
+  (oset object :hash (nix-store--query 'hash (nix-store-path-path object)))
+  (oset object :referrers (nix-store--query 'referrers (nix-store-path-path object)))
+  (oset object :requisites (nix-store--query 'requisites (nix-store-path-path object)))
+  (oset object :references (nix-store--query 'references (nix-store-path-path object)))
+  (oset object :derivers (nix-store--query 'deriver (nix-store-path-path object)))
+  object)
+
+(defun nix-store--query (argument path)
+  (let ((nix-executable nix-store-executable))
+    (cond
+      ((eq 'deriver argument) (nix--process-lines "--query" "--deriver" path ))
```

Comment:
it would be cool to hyperlink this to a derivation view

Reply:
Oh I see C-c C-o now. I was expected enter / click to work though.

Reply:
When a path ends with .drv, we probably want to skip deriver. It's always going to be "unknown".

Reply:
> Oh I see C-c C-o now. I was expected enter / click to work though.

Yes, `RET` is more intuitive. I have changed it to that.

> When a path ends with .drv, we probably want to skip deriver. It's always going to be "unknown".

Done.